dash/dash.py

```diff
@@ -805,6 +811,23 @@ def wrap_func(func):
             def add_context(*args, **kwargs):
 
                 output_value = func(*args, **kwargs)
+                setattr(
+                    self._components[output.component_id],
+                    output.component_property,
+                    output_value
+                )
+                self._components[output.component_id]._validator.validate({
+                    output.component_property: output_value
+                })
+                if output.component_property == 'children':
```

Comment:
If the component prop is `children`, we must also add any new components with `id` fields which may have been added to the layout as a result of this callback to the `self._components` dictionary.